### DISCUSSION TITLE
Enable GPUExternalTexture on WebGPU?

### DISCUSSION THREAD

**Discussion by seancribbs** [gfx-rs/wgpu#3029] [Q&A] [answered]
In the future, we want to be able to render video frames into our scene in the browser. I noticed that the WebGPU spec has a type [`GPUExternalTexture`](https://www.w3.org/TR/webgpu/#gpu-external-texture) that can be extracted from a `<video>` element, but there's no analog in [`wgpu::BindingType`](https://docs.rs/wgpu/latest/wgpu/enum.BindingType.html).

Obviously, there are limitations in exposing backend-specific APIs while remaining platform-neutral, but how might one go about adding this feature specifically for the WebGPU backend? Can we do it with conditional compilation or is there some existing API we can leverage? Would a pull-request implementing this feature be appreciated or is it low on the priority list?

**Comment by i509VCB** [comment, marked as answer]
Hi, I've opened #3145 related to this issue. It would cover support for video elements on web.
